lib/Queue.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const { emitWarning } = require('process')
+
+/**
+ * @typedef {object} QueueOptions
+ * @prop {number} [maxSize] Positive integer to track the sizes of items added to the cache, and automatically evict items in order to stay below this size. Default: `Infinity`
+ * @prop {(any) => number?} [sizeCalculation] Function used to calculate the size of stored elements. Default: `element => element.length`.
+ */
+
+/**
+ * A linear data structure of a FIFO (First In - First Out) type.
+ *
+ * Inspired by:
+ * - https://vhudyma-blog.eu/implement-queue-in-javascript/
+ * - https://github.com/isaacs/node-lru-cache
+ */
+module.exports = class Queue {
+  /**
+   * @param {QueueOptions} opts
+   */
+  constructor (opts) {
+    // The actual queue
+    this.elements = {}
+    // The index of the head element
+    this.head = 0
+    // The index of the tail element
+    this.tail = 0
+    this.calculatedSize = 0
+    this.sizes = {}
+    this.opts = {
+      maxSize: Infinity,
```

Comment:
Let's make this 1024 instead, we don't want applications that have memory leaks.